lib/SyTest/Synapse.pm

```diff
@@ -239,6 +239,37 @@ sub start
             )
          );
 
+         my $polling_period = 0.01;
+         my $polling_count = 20;
+         my $poll;
+         $poll = sub {
```

Comment:
This is the closure self-capturing antipattern that `Future::Utils::repeat` is designed to help you avoid. You probably want to use that instead.

(the primary reason being it memory-leaks; though not a major disaster in this case because it's once at startup)